tests/providers/ftp/operators/test_ftp.py

```diff
@@ -184,3 +184,112 @@ def test_unequal_local_remote_file_paths(self):
+        assert not mock_create_dir.called
+        mock_put.assert_called_once_with(self.test_remote_filepath, self.test_local_filepath)
+
+    @mock.patch("airflow.providers.ftp.operators.ftp.FTPSHook.store_file")
+    @mock.patch("airflow.providers.ftp.operators.ftp.FTPSHook.create_directory")
+    def test_file_transfer_with_intermediate_dir_put(self, mock_create_dir, mock_put):
+        ftps_op = FTPSFileTransmitOperator(
+            task_id="test_ftps_put_imm_dirs",
+            ftp_conn_id=DEFAULT_CONN_ID,
+            local_filepath=self.test_local_filepath,
+            remote_filepath=self.test_remote_filepath_int_dir,
+            operation=FTPOperation.PUT,
+            create_intermediate_dirs=True,
+        )
+        ftps_op.execute(None)
+        mock_create_dir.assert_called_with(self.test_remote_dir_int)
+        mock_put.assert_called_once_with(self.test_remote_filepath_int_dir, self.test_local_filepath)
+
+    @mock.patch("airflow.providers.ftp.operators.ftp.FTPSHook.retrieve_file")
+    def test_file_transfer_get(self, mock_get):
+        ftps_op = FTPSFileTransmitOperator(
+            task_id="test_ftps_get",
+            ftp_conn_id=DEFAULT_CONN_ID,
+            local_filepath=self.test_local_filepath,
+            remote_filepath=self.test_remote_filepath,
+            operation=FTPOperation.GET,
+        )
+        ftps_op.execute(None)
+        mock_get.assert_called_once_with(self.test_remote_filepath, self.test_local_filepath)
+
+    @mock.patch("airflow.providers.ftp.operators.ftp.FTPSHook.retrieve_file")
+    def test_file_transfer_with_intermediate_dir_get(self, mock_get):
+        ftps_op = FTPSFileTransmitOperator(
+            task_id="test_ftps_get_imm_dirs",
+            ftp_conn_id=DEFAULT_CONN_ID,
+            local_filepath=self.test_local_filepath_int_dir,
+            remote_filepath=self.test_remote_filepath,
+            operation=FTPOperation.GET,
+            create_intermediate_dirs=True,
+        )
+        ftps_op.execute(None)
+        assert os.path.exists(self.test_local_dir_int)
+        mock_get.assert_called_once_with(self.test_remote_filepath, self.test_local_filepath_int_dir)
+
+    @mock.patch("airflow.providers.ftp.operators.ftp.FTPSHook.retrieve_file")
+    def test_multiple_paths_get(self, mock_get):
+        local_filepath = ["/tmp/ltest1", "/tmp/ltest2"]
+        remote_filepath = ["/tmp/rtest1", "/tmp/rtest2"]
+        ftps_op = FTPSFileTransmitOperator(
+            task_id="test_multiple_paths_get",
+            ftp_conn_id=DEFAULT_CONN_ID,
+            local_filepath=local_filepath,
+            remote_filepath=remote_filepath,
+            operation=FTPOperation.GET,
+        )
+        ftps_op.execute(None)
+        assert mock_get.call_count == 2
+        args0, _ = mock_get.call_args_list[0]
+        args1, _ = mock_get.call_args_list[1]
+        assert args0 == (remote_filepath[0], local_filepath[0])
+        assert args1 == (remote_filepath[1], local_filepath[1])
```

Comment:
Here and below, just a suggestion/nitpick, but maybe slightly easier as:

```
expected_call_count = 2
assert mock_get.call_count == expected_call_count
for call in range(expected_calls_count):
    assert mock_get.call_args_list[call][0] == (remote_filepath[call], local_filepath[call])
```

Reply:
@ferruzzi I've added this change as well. Let me know if it looks good and if any other change is needed.